packages/yourloops/lib/team/utils.tsx

```diff
@@ -77,92 +72,55 @@ export default class TeamUtils {
     }, 0)
   }
 
-  static iMemberToMember(iTeamMember: ITeamMember, team: Team, users: Map<string, TeamUser>): TeamMember {
+  static iMemberToMember(iTeamMember: ITeamMember, teamId: string, invitations: INotification[]): TeamMember {
     const {
       userId,
       invitationStatus,
       role,
       email,
-      preferences,
-      profile,
-      settings,
-      idVerified,
-      alarms,
-      monitoring,
-      unreadMessages
+      profile
     } = iTeamMember
-
-    let teamUser = users.get(userId)
-    if (!teamUser) {
-      teamUser = {
-        role: role === TeamMemberRole.patient ? UserRoles.patient : UserRoles.hcp,
-        userid: userId,
-        username: email,
-        emails: [email],
-        preferences,
-        profile,
-        settings: fixYLP878Settings(settings),
-        members: [],
-        idVerified,
-        alarms,
-        monitoring,
-        unreadMessages
-      }
-      users.set(userId, teamUser)
-    }
-
-    const teamMember: TeamMember = {
-      team,
+    return {
+      userId,
+      email,
+      profile,
       role,
       status: invitationStatus,
-      user: teamUser
+      invitation: invitations.find(invitation => invitation.target.id === teamId && invitation.email === email)
     }
-    teamUser.members.push(teamMember)
-    team.members.push(teamMember)
-    return teamMember
   }
 
-  static iTeamToTeam(iTeam: ITeam, users: Map<string, TeamUser>): Team {
-    const team: Team = { ...iTeam, members: [] }
-    // Detect duplicate users, and update the member if needed
-    iTeam.members.forEach(iTeamMember => TeamUtils.iMemberToMember(iTeamMember, team, users))
-    return team
-  }
-
-  static getUserByEmail(teams: Team[], email: string): TeamUser | null {
-    for (const team of teams) {
-      for (const member of team.members) {
-        if (member.user.username === email) {
-          return member.user
-        }
-      }
-    }
-    return null
-  }
-
-  static async loadTeams(user: User): Promise<Team[]> {
-    const users = new Map<string, TeamUser>()
-    const apiTeams = await TeamApi.getTeams()
-
-    log.debug('loadTeams', { nTeams: apiTeams.length })
+  static async loadTeams(userId: string, invitations: INotification[]): Promise<Team[]> {
+    const iTeams = await TeamApi.getTeams()
 
     const privateTeam: Team = {
```

Comment:
Why do we create a privateTeam ?

Reply:
From what I know, this is useful for directshares

Reply:
ok !